Tabloid/client/src/modules/categoryManager.js

```diff
@@ -0,0 +1,80 @@
+import { getToken } from "./authManager";
+
+const baseUrl = "/api/category";
+
+export const getAllCategories = () => {
+  return getToken().then((token) =>
+    fetch(`${baseUrl}`, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        throw new Error("Error");
+      }
+    })
+  );
+};
+
+export const addCategory = (category) => {
+  return getToken().then((token) =>
+    fetch(baseUrl, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify(category),
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        throw new Error("Error");
+      }
+    })
+  );
+};
+
+export const deleteCategory = (id) => {
```

Comment:
Should this function include some kind of error handling?

Reply:
Updated to have error handling. Once I commit, you will see the changes.